tests/memory_checker/test_memory_checker.py

```diff
@@ -206,7 +214,12 @@ def test_memory_checker(duthosts, creds, enum_rand_one_per_hwsku_frontend_hostna
     Returns:
         None.
     """
-    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    dut_name, container_name = decode_dut_port_name(enum_dut_feature_container)
+    pytest_require(dut_name == enum_rand_one_per_hwsku_frontend_hostname and container_name == "telemetry",
+                   "Skips testing memory_checker of container '{}' on the DuT '{}' since another DuT '{}' was chosen."
+                   .format(container_name, dut_name, enum_rand_one_per_hwsku_frontend_hostname))
+    duthost = duthosts[dut_name]
```

Comment:
@bingwang-ms Regarding to the question we can use the `duthost = rand_selected_dut` to replace the line 221, I tested it did work. Also this test case currently is only running against `telemetry` container. 

However, there is an issue. Suppose we have Dual ToR `str2-7050cx3-acs-06` and `str2-7050cx3-acs-07`, The parameterized fixture `enum_dut_feature_container` includes a list of strings such as `str2-7050cx3-acs-06|lldp`, `str2-7050cx3-acs-06|database`, `str2-7050cx3-acs-06|snmp`, `str2-7050cx3-acs-06|telemetry`, ..., `str2-7050cx3-acs-07|lldp`, `str2-7050cx3-acs-07|database`, `str2-7050cx3-acs-07|snmp`, `str2-7050cx3-acs-07|telemetry`, ....

The function `decode_dut_port_name(...)` will parse each string to separate the DuT name from container name. If we use `duthost = rand_selected_dut` to replace the line 221 and `rand_selected_dut` is `str2-7050cx3-acs-06`, then this test case will run two times in `telemetry` container on DuT `str2-7050cx3-acs-06`. One time it is running for input string `str2-7050cx3-acs-06|telemetry` and another is for `str2-7050cx3-acs-07|telemetry`. 

Overall I think Ying's method using `pytest_require(dut_name == enum_rand_one_per_hwsku_frontend_hostname, ...)` does make sense.

Reply:
Why not add a filter to the output of ```decode_dut_port_name``` (actually ```decode_dut_feature```)? Checking  ```pytest_require(dut_name == enum_rand_one_per_hwsku_frontend_hostname, ...)``` will cause test randomly run or skip. I don't think it's expected.